front/components/agent_builder/capabilities/mcp/formValidation.ts

```diff
@@ -0,0 +1,189 @@
+import { z } from "zod";
+
+import type { MCPServerConfigurationType } from "@app/components/agent_builder/AgentBuilderFormContext";
+import {
+  additionalConfigurationSchema,
+  childAgentIdSchema,
+  dustAppConfigurationSchema,
+  jsonSchemaFieldSchema,
+  jsonSchemaStringSchema,
+  mcpServerViewIdSchema,
+  mcpTimeFrameSchema,
+  reasoningModelSchema,
+  tablesConfigurationsSchema} from "@app/components/agent_builder/AgentBuilderFormContext";
+import type { MCPServerRequirements } from "@app/lib/actions/mcp_internal_actions/input_configuration";
+import { getMCPServerRequirements } from "@app/lib/actions/mcp_internal_actions/input_configuration";
+import type { MCPServerViewType } from "@app/lib/api/mcp";
+
+const createConfigSchema = (requirements: MCPServerRequirements | null) => {
+  const baseFields = {
+    mcpServerViewId: mcpServerViewIdSchema,
+    timeFrame: mcpTimeFrameSchema,
+    jsonSchema: jsonSchemaFieldSchema,
+    _jsonSchemaString: jsonSchemaStringSchema,
+  };
+
+  if (!requirements) {
+    return z.object({
+      ...baseFields,
+      dataSourceConfigurations: z.any().nullable().default(null), // TODO: fixme
+      tablesConfigurations: tablesConfigurationsSchema,
+      childAgentId: childAgentIdSchema,
+      reasoningModel: reasoningModelSchema,
+      dustAppConfiguration: dustAppConfigurationSchema,
+    });
+  }
+
+  const dynamicFields: Record<string, z.ZodSchema> = {};
+
+  if (requirements.requiresChildAgentConfiguration) {
+    dynamicFields.childAgentId = childAgentIdSchema.refine(
+      (val) => val !== null,
+      {
+        message: "Child agent selection is required",
+      }
+    );
+  } else {
+    dynamicFields.childAgentId = z.null();
+  }
+
+  if (requirements.requiresReasoningConfiguration) {
+    dynamicFields.reasoningModel = reasoningModelSchema.refine(
+      (val) => val !== null,
```

Comment:
I just check if there is a value or not and I don't think we need to check every field?